### PR TITLE
Fix various issues with product analytics

### PR DESCRIPTION
**Pull request by johnschoeman**
Why:
Currently we have a handful of visual and logic bugs in the product
analytics feature.

This commit:
- Verifies if the correct Matomo variables have been provided before
initializing the Matomo SDK
- moves the logic of determining if the build should enable Product
analytics out of the Product analytics context to the app initialization
to reduce complexity in the context.
- Separates the analytics consent screen into two components, one for
the onboarding activation flow and one for the settings screen to reduce
complexity in the navigation logic and reduce complexity in the forms
themselves.
- explicitly states if product analytics should be enabled rather than
inferring from the presence of the Matomo url
- removes the attempted tracking of a user opting into product analytics
as its current behavior only tracks when a user opts out of sharing
data, but still labels it opting in. We will add this back in a future
commit.
- removes the tracking call of if a user received an exposure
notification as this caused crashes in some cases. We will add this back
in a future commit.

### CODE REVIEW

File: App.tsx [hunk 1]
```diff
@@ -45,21 +45,39 @@ const App: FunctionComponent = () => {
   }, [])
 
   const initializeMatomo = () => {
-    const displayAnalytics = env.DISPLAY_ANALYTICS
     const matomoUrl = env.MATOMO_URL
     const matomoSiteId = parseInt(env.MATOMO_SITE_ID)
-    if (displayAnalytics) {
+
+    if (matomoUrl && !isNaN(matomoSiteId)) {
```
Comment (devinjameson) [inline App.tsx]: Is it worth giving `!isNaN(matomoSiteId)` a name?

File: App.tsx [hunk 2]
```diff
@@ -45,21 +45,39 @@ const App: FunctionComponent = () => {
...
+    const nullClient = {
+      trackEvent: async (_category: string, _action: string) => {},
+      trackView: async (_routes: string[]) => {},
+    }
 
-  const productAnalyticsClient: ProductAnalyticsClient = {
-    trackEvent: Matomo.trackEvent,
-    trackView: Matomo.trackView,
+    if (displayAnalytics) {
+      const client = initializeMatomo()
```
Comment (devinjameson) [inline App.tsx:73]: 💯

File: src/Activation/ProductAnalyticsConsentForm.tsx [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+  TouchableOpacity,
+} from "react-native"
+import { useNavigation } from "@react-navigation/native"
+import { useTranslation } from "react-i18next"
+
+import { useConfigurationContext } from "../ConfigurationContext"
+import { useProductAnalyticsContext } from "../ProductAnalytics/Context"
+import { ActivationStackScreens, useStatusBarEffect } from "../navigation"
+
+import { Text } from "../components"
```
Comment (devinjameson) [inline src/Activation/ProductAnalyticsConsentForm.tsx:15]: I think stuff from `components` may want to live in the second group of imports.

File: src/Activation/ProductAnalyticsConsentForm.tsx [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+  const handleOnPressMaybeLater = () => {
+    navigation.navigate(ActivationStackScreens.ActivationSummary)
+  }
+
+  const handleOnPressPrivacyPolicy = () => {
+    Linking.openURL(healthAuthorityPrivacyPolicyUrl)
+  }
+  const privacyPolicyLinkText = t("product_analytics.privacy_policy")
+
+  const buttonText = t("product_analytics.i_understand_and_consent")
```
Comment (devinjameson) [inline src/Activation/ProductAnalyticsConsentForm.tsx:43]: Is there a reason these bits of text are pulled out?

**Review by devinjameson — APPROVED** [review]
Great work!

File: src/Activation/ProductAnalyticsConsentForm.tsx (hunk 4 above)
Comment (johnschoeman) [reply src/Activation/ProductAnalyticsConsentForm.tsx:43]: artifact of the refactor, can inline.

File: App.tsx (hunk 1 above)
Comment (johnschoeman) [reply App.tsx]: i think so!

**Review by mattThousand — APPROVED** [review]

**Pull request merged**